pkg/health/health.go

```diff
@@ -0,0 +1,75 @@
+package health
+
+import (
+	"context"
+	"time"
+
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/metrics"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/pkg/errors"
+)
+
+// Checker validates that revision is healthy.
+type Checker interface {
+	IsHealthy(ctx context.Context, query metrics.Query, healthCheckOffset time.Duration, metricsChecks []config.Metric) (bool, error)
+}
+
+// Health is the candidate's health status checker.
+type Health struct {
+	Metrics metrics.Metrics
+}
+
+// New initializes a health checker.
+func New(metrics metrics.Metrics) *Health {
+	return &Health{
+		Metrics: metrics,
+	}
+}
+
+// IsHealthy checks if a revision is healthy based on the metrics configuration.
+func (h *Health) IsHealthy(ctx context.Context, query metrics.Query, healthCheckOffset time.Duration, metricsChecks []config.Metric) (bool, error) {
```

Comment:
Maybe we need to do this in another PR:

If you just return `bool` for fail/pass, we cannot capture details like "what metrics did we see", "why did we make this decision". 

As part of our design we said we'll provide these insights back to the users. So probably we need to export something like []CheckResults that explains result of each criteria checked and if it was "met" or "not".